### ISSUE TITLE
`attention_mask` is not passed into the model during training?

### ISSUE THREAD

**Issue by SamithVa** [showlab/ShowUI#78]
If we don't pass an explicit attention mask during training, and the model generates one internally using `_prepare_4d_causal_attention_mask_with_cache_position`, how does it know which tokens are padding when the batch size is greater than 1?

https://github.com/showlab/ShowUI/blob/2e71e6a8421b38434526b4c95523812d8abab8e1/data/dataset.py#L45C1-L45C68

**Comment by QinghongLin** [comment]
Hi @SamithVa 
Yes, I believe this is no needed, as so far the training codebase is implemented by batchsize=1, so attention mask will be automatically calculated based on input_ids / values. Thank you!
